cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java

```diff
@@ -569,15 +608,145 @@ private boolean isToolLocallyAvailable(String toolId, String version) {
         if (version.equals(Names.EMPTY.getValue())) {
             return false;
         }
-        BalToolsToml distSpecificToolsToml = BalToolsToml.from(distSpecificToolsTomlPath);
-        BalToolsManifest distSpecificToolsManifest = BalToolsManifestBuilder.from(distSpecificToolsToml).build();
-        if (distSpecificToolsManifest.tools().containsKey(toolId)) {
-            BalToolsManifest.Tool tool = distSpecificToolsManifest.tools().get(toolId);
-            Path toolPath = Path.of(
-                    System.getProperty(CommandUtil.USER_HOME), HOME_REPO_DEFAULT_DIRNAME, REPOSITORIES_DIR,
-                    CENTRAL_REPOSITORY_CACHE_NAME, BALA_DIR_NAME, tool.org(), tool.name(), version);
-            return toolPath.toFile().exists();
+        BalToolsToml balToolsToml = BalToolsToml.from(balToolsTomlPath);
+        BalToolsManifest balToolsManifest = BalToolsManifestBuilder.from(balToolsToml).build();
+        Optional<BalToolsManifest.Tool> tool = balToolsManifest.getTool(toolId, version);
+        if (tool.isEmpty()) {
+            return false;
+        }
+        Path toolCacheDir = ProjectUtils.createAndGetHomeReposPath()
+                .resolve(REPOSITORIES_DIR).resolve(CENTRAL_REPOSITORY_CACHE_NAME)
+                .resolve(ProjectConstants.BALA_DIR_NAME).resolve(tool.get().org()).resolve(tool.get().name());
+        if (toolCacheDir.toFile().isDirectory()) {
+            try (Stream<Path> versions = Files.list(toolCacheDir)) {
+                return versions.anyMatch(path -> {
```

Comment:
Let's simplify with something like
 ...           
            if (versions.anyMatch(path -> path.getFileName().toString().equals(version))) {
                org = tool.get().org();
                name = tool.get().name();
                return true;
            }
....

and let's assign `tool.get()` to a variable and use

Let's make sure to exit early whenever possible